app/src/main/kotlin/com/wire/android/ui/calling/SharedCallingViewModel.kt

```diff
@@ -154,12 +154,15 @@ class SharedCallingViewModel @Inject constructor(
     }
 
     private suspend fun observeOnMute() {
-        snapshotFlow { callState.isMuted }.collectLatest {
-            it?.let {
-                if (it) {
-                    muteCall(conversationId)
-                } else {
-                    unMuteCall(conversationId)
+        //We should only mute established calls
+        snapshotFlow { callState.isMuted to callState.callStatus }.collectLatest { isMute ->
+            if (isMute.second == CallStatus.ESTABLISHED) {
+                isMute.first?.let {
+                    if (it) {
+                        muteCall(conversationId)
+                    } else {
+                        unMuteCall(conversationId)
+                    }
                 }
             }
```

Comment:
```suggestion
          snapshotFlow { callState.isMuted to callState.callStatus }.collectLatest { (isMuted, callStatus) ->
            if (callStatus == CallStatus.ESTABLISHED) {
                isMuted?.let {
                    if (it) {
                        muteCall(conversationId)
                    } else {
                        unMuteCall(conversationId)
                    }
                }
            }
        }
```

I believe something like this, using the deconstructing assignment could help a bit to avoid the `.first` and `.second` calls